### PR TITLE
[PyTorch] Fix Test's Reference Implementation of Clamped Swiglu

### PR DESCRIPTION
**Pull request by vthumbe1503**
# Description

Pytorch recently changed their clamp function's autograd behavior in this PR: https://github.com/pytorch/pytorch/pull/191142/changes#diff-e1a04c95c2809af34915952d3c7cc6210fadd2a277af1cfdb4924c74a5795295R121

torch.clamp method's original behavior was derivative = 1 at the clamp boundary. Now they changed it to be 0 instead.(I think one of the reasons is consistency of relu operation which has also produces zero derivative at boundary=0)

This PR changes the reference implementation of clamped swiglu in pytorch to use torch.where() to explicitly define the boundary gradient to be 0.

Fixes # (issue)

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refactoring

## Changes

Please list the changes introduced in this PR:

- Change A
- Change B

# Checklist:

- [ ] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [ ] The functionality is complete
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/pytorch/test_fusible_ops.py [hunk 1]
```diff
@@ -2550,8 +2550,9 @@ def test_clamped_swiglu(
             x = x.transpose(-3, -2)
             x = x.reshape(in_shape)
         x_glu, x_linear = x.chunk(2, dim=-1)
-        x_glu = x_glu.clamp(min=None, max=limit)
-        x_linear = x_linear.clamp(min=-limit, max=limit)
```
Comment (timmoon10) [inline tests/pytorch/test_fusible_ops.py:2554]: ~I assume the difference is that clamp kills the gradients at the limits, while our activation functions preserve them? But we've always passed gradients at the clamped SwiGLU limits (see https://github.com/NVIDIA/TransformerEngine/pull/2161), so that doesn't explain why the recent test failures.~

PyTorch recently changed the behavior of clamp to kill grads at the limits: https://github.com/pytorch/pytorch/pull/191142/changes#diff-e1a04c95c2809af34915952d3c7cc6210fadd2a277af1cfdb4924c74a5795295R121

**Review by timmoon10 — COMMENTED** [review]
~Do we know why this error started appearing?~

~I'm suspicious of https://github.com/NVIDIA/TransformerEngine/pull/3402, which touched the logic for GLU backwards. However, the existing logic also preserved gradients at boundaries. Also, the CI passed for that PR.~

~https://github.com/NVIDIA/TransformerEngine/pull/3359 landed exactly when the CI pipelines started failing, but it doesn't seem relevant and it passed CI.~

I see that the errors are caused by [a recent change in PyTorch](https://github.com/pytorch/pytorch/pull/191142/changes#diff-e1a04c95c2809af34915952d3c7cc6210fadd2a277af1cfdb4924c74a5795295R121). This is a reasonable fix.

**Review by timmoon10 — APPROVED** [review]

**Comment by vthumbe1503** [comment]
Errors are pre-existing in main
